pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroRecordExtractor.java

```diff
@@ -31,22 +34,22 @@
 import org.apache.pinot.spi.data.readers.GenericRow;
 import org.apache.pinot.spi.data.readers.RecordExtractorConfig;
 
-
-/**
- * Extractor for Avro Records
- */
+/** Extractor for Avro Records */
 public class AvroRecordExtractor extends BaseRecordExtractor<GenericRecord> {
   private Set<String> _fields;
   private boolean _extractAll = false;
+  private boolean _applyLogicalTypes = false;
```

Comment:
this setting should come from table config in order to potentially break existing ingestion that come with logical types, however `RecordExtractorConfig` is always null. Is anyone working on this logic at the moment? I could give that a go as I'll need to add the setting for this feature

Reply:
yes I think we can create a `AvroRecordExtractorConfig` and provide some defaults to this if necessary. see `CsvRecordExtractorConfig` as an example. i think it can be part of this PR